protocol/transaction_validator.go

```diff
@@ -0,0 +1,289 @@
+package protocol
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/hex"
+	"fmt"
+	"github.com/pkg/errors"
+	coretypes "github.com/status-im/status-go/eth-node/core/types"
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"math/big"
+	"strings"
+)
+
+const (
+	transferFunction        = "a9059cbb"
+	tokenTransferDataLength = 68
+	transactionHashLength   = 66
+)
+
+type TransactionValidator struct {
+	persistence *sqlitePersistence
+	addresses   map[string]bool
```

Comment:
If checking only existence:
```suggestion
	addresses   map[string]struct{}
```